sktwocnt/lib/twocnt.f90

```diff
@@ -463,76 +463,45 @@ subroutine getskintegrals(t_integ, nRad, nAng, atom1, atom2, grid1, grid2, dots,
       radval2pp(:, ii) = atom2%ddrad(ii)%getValue(r2)
     end do
 
-    allocate(potval(nGrid))
     ifPotSup: if (.not. tDensitySuperpos) then
-      potval(:) = atom1%pot%getValue(r1) + atom2%pot%getValue(r2)
+      potval = atom1%pot%getValue(r1) + atom2%pot%getValue(r2)
     else
       allocate(densval(nGrid))
       densval(:) = atom1%rho%getValue(r1) + atom2%rho%getValue(r2)
 
       ! prepare xc-functional specific arrays
-      select case(iXC)
-      case(2)
-        ! LDA needs just density
-      ! case(3:4) with in-house code
-      case(3:4)
-        allocate(densval1p(nGrid))
-        allocate(densval1pp(nGrid))
-        allocate(densval2p(nGrid))
-        allocate(densval2pp(nGrid))
-        densval1p = atom1%drho%getValue(r1)
-        densval1pp = atom1%ddrho%getValue(r1)
-        densval2p = atom2%drho%getValue(r2)
-        densval2pp = atom2%ddrho%getValue(r2)
-        allocate(absgr(nGrid))
-        allocate(laplace(nGrid))
-        allocate(gr_grabsgr(nGrid))
-        ! calculate derivatives for combined density
-        call getDerivs(densval1p, densval1pp, densval2p, densval2pp, r1, r2, dots, absgr, laplace,&
-            & gr_grabsgr)
-        deallocate(densval1p, densval1pp, densval2p, densval2pp)
-      ! libXC treatment requires sigma = (nabla n).(nabla n) for GGA/LC
-      case(5:8)
-        allocate(densval1p(nGrid))
-        allocate(densval2p(nGrid))
+      ! care about correct 4pi normalization of density
+      rhor = getLibxcRho(densval)
+      allocate(vx(nGrid))
+      allocate(vc(nGrid))
+      if (iXC /= 1) then
```

Comment:
use the type-based enum value also here.

Reply:
Good point :)